spring-kafka/src/test/java/org/springframework/kafka/support/serializer/RetryingDeserializerTests.java

```diff
@@ -54,6 +55,39 @@ void testRetry() {
 		rdes.close();
 	}
 
+	@Test
+	void retryDeserializerWithRecoveryCallback() throws InterruptedException {
+		NonRecoverableDeser delegate = new NonRecoverableDeser();
+		RetryingDeserializer<String> rdes = new RetryingDeserializer<>(delegate, new RetryTemplate());
+		rdes.setRecoveryCallback(context -> {
+			delegate.latch.countDown();
+			return null;
+		});
+		assertThat(rdes.deserialize("my-topic", "my-data".getBytes())).isNull();
+		assertThat(delegate.latch.await(10, TimeUnit.SECONDS)).isTrue();
+		rdes.close();
+	}
+
+	public static class NonRecoverableDeser implements Deserializer<String> {
+
+		CountDownLatch latch = new CountDownLatch(4);
+
+		@Override
+		public void configure(Map<String, ?> configs, boolean isKey) {
+		}
+
+		@Override
+		public String deserialize(String topic, byte[] data) {
+			latch.countDown();
+			throw new RuntimeException();
+		}
+
+		@Override
+		public void close() {
```

Comment:
The `Deserializer` is functional interface, so no need to override those `default` methods.
See my comment above about its simple lambda for the `RetryingDeserializer` ctor.